tests/jobs/localisation/example_case/scripts/init_test_case.py

```diff
@@ -0,0 +1,262 @@
+        for i, indx in enumerate(obs_index_list):
+            Iindx, Jindx, Kindx = indx
+            obs_name = f"OBS_{Iindx}_{Jindx}_{Kindx}"
+            pos = positions[i]
+            file.write(f"{space}- name: CORR_{i}\n")
+            file.write(f"{space2}obs_group:\n")
+            file.write(f'{space3}add: ["{obs_name}"]\n')
+            file.write(f"{space2}param_group:\n")
+            file.write(f'{space3}add: ["{field_name}"]\n')
+            file.write(f"{space2}field_scale:\n")
+            file.write(f"{space3}method: gaussian_decay\n")
+            file.write(f"{space3}main_range: {corr_ranges[0]}\n")
+            file.write(f"{space3}perp_range: {corr_ranges[1]}\n")
+            file.write(f"{space3}azimuth: {azimuth}\n")
+            file.write(f"{space3}ref_point: [ {pos[0]}, {pos[1]} ]\n")
+
+
+def write_gen_obs(upscaled_values):
+    observation_dir = settings.observation.directory
+    obs_file_name = settings.observation.file_name
+    obs_data_dir = settings.observation.data_dir
+    cell_indx_list = settings.observation.selected_grid_cells
+    rel_err = settings.observation.rel_error
+    min_err = settings.observation.min_abs_error
+    if not os.path.exists(observation_dir):
+        print(f"Create directory: {observation_dir} ")
+        os.makedirs(observation_dir)
+    data_dir = observation_dir + "/" + obs_data_dir
+    if not os.path.exists(data_dir):
+        print(f"Create directory: {data_dir} ")
+        os.makedirs(data_dir)
+
+    print(f"Write observation file: {obs_file_name} ")
+    filename = observation_dir + "/" + obs_file_name
+    with open(filename, "w", encoding="utf8") as obs_file:
+        number = 0
+        for indices in cell_indx_list:
+            Iindx = indices[0] - 1
+            Jindx = indices[1] - 1
+            Kindx = indices[2] - 1
+
+            value = upscaled_values[Iindx, Jindx, Kindx]
+            value_err = math.fabs(value) * rel_err
+            value_err = max(value_err, min_err)
+
+            obs_data_relative_file_name = (
+                obs_data_dir
+                + "/obs_"
+                + str(Iindx + 1)
+                + "_"
+                + str(Jindx + 1)
+                + "_"
+                + str(Kindx + 1)
+                + ".txt"
+            )
+
+            obs_file.write(f"GENERAL_OBSERVATION   OBS_{Iindx+1}_{Jindx+1}_{Kindx+1}  ")
+            obs_file.write("{ ")
+            obs_file.write(
+                f"DATA = RESULT_UPSCALED_FIELD ; INDEX_LIST = {number} ; RESTART = 0;  "
+            )
+            obs_file.write(f"OBS_FILE = ./{obs_data_relative_file_name} ; ")
+            obs_file.write(" };\n")
+            number += 1
+            data_file_name = observation_dir + "/" + obs_data_relative_file_name
+            print(f"Write file: {data_file_name} ")
+            with open(data_file_name, "w", encoding="utf8") as data_file:
+                data_file.write(f"{value}  {value_err}\n")
+
+
+def create_grid():
```

Comment:
The `create_grid` and other functions rely on a global settings object for their configuration. It makes the code harder to understand I think.
Could we instead consider parameterizing the functions?
For `create_grid` in particular, we could perhaps do something like this:

```python
def create_grid(grid_file_name, grid_dimension, grid_size, use_eclipse_grid_index_origo=False):
```

Reply:
Done